### PR TITLE
Add maintenance mode toggle support

### PR DESCRIPTION
**Pull request by flashadvocate**

### CODE REVIEW

File: manage.py [hunk 1]
```diff
@@ -311,6 +313,27 @@ def update_node_packages(project_config):
...
+    artisan_path = os.path.join(project_path, 'artisan')
+    maintenance_file = os.path.join(project_path, 'storage', 'framework', 'maintenance.php')
+
+    if not os.path.exists(artisan_path):
+        print(f"Error: Maintenance not supported for this project. Operation aborted.")
+        sys.exit(1)
+
+    action = 'up' if os.path.exists(maintenance_file) else 'down'
+
+    command = f"sudo -u {deploying_user} php {artisan_path} {action}"
```
Comment (apschultz) [inline manage.py]: Should this be done in the container to avoid php compat issues?

Comment (flashadvocate) [reply manage.py]: Yes, actually. Ugh it’s been a day

File: manage.py [hunk 2]
```diff
@@ -311,6 +313,27 @@ def update_node_packages(project_config):
     except Exception as e:
         print("An error occurred during Node.js package update: " + str(e))
 
+def toggle_maintenance_mode(project_config):
+    project_path = project_config['path']
```
Comment (apschultz) [inline manage.py:301]: Can we also check for existence of params? The exception output is a pain to trap.

Looks like path and container need to be checked.

Comment (flashadvocate) [reply manage.py:301]: @apschultz did one better and cleaned up the conditionals in favor of a dedicated config_check method in https://github.com/ClanAODDev/deploy/pull/6/commits/a365d84e74f697346eb51c53f3980808211c55a5

**Review by apschultz — APPROVED** [review]

**Pull request merged**